src/wasm-type.h

```diff
@@ -195,10 +195,31 @@ struct Signature {
 
 struct Field {
   Type type;
+  enum PackedType : uint32_t {
+    not_packed,
+    i8,
+    i16,
+  } packedType; // applicable iff type=i32
   bool mutable_;
-  Field(Type type, bool mutable_ = false) : type(type), mutable_(mutable_) {}
+
+  Field(Type type, bool mutable_ = false)
+    : type(type), packedType(not_packed), mutable_(mutable_) {}
+  Field(Type::BasicID type, bool mutable_ = false)
+    : type(type), packedType(not_packed), mutable_(mutable_) {}
```

Comment:
Added this to avoid ambiguity between `Type` and `PackedType` given as uints, but I don't actually know what I'm doing there. How bad is it? :)

Reply:
Is this because the compiler complains about ambiguity between `Type` and `uint32_t` when you do something like `Field(Type::i32)`? If so, adding this extra constructor to resolve the ambiguity seems fine 👍

Reply:
Didn't test (should have), but tested it now and there doesn't seem to be any problem without it. The compiler always complains when providing an `uint32_t` anyway, and I was suspecting that with the additional constructor, it would hint at the ambiguity instead of silently and wrongly picking `PackedType`, which isn't the case. Anyway, thanks! :)